DevProxy.Plugins/Reporting/MinimalPermissionsGuidancePluginReport.cs

```diff
@@ -0,0 +1,179 @@
+                .AppendLine("### Requests")
+                .AppendLine()
+                .AppendJoin(Environment.NewLine, result.Requests.Select(r => $"- {r}"))
+                .AppendLine()
+                .AppendLine()
+                .AppendLine("### Minimal permissions")
+                .AppendLine()
+                .AppendJoin(Environment.NewLine, result.MinimalPermissions.Select(p => $"- `{p}`"))
+                .AppendLine()
+                .AppendLine()
+                .AppendLine("### Permissions on the token")
+                .AppendLine()
+                .AppendJoin(Environment.NewLine, result.TokenPermissions.Select(p => $"- `{p}`"))
+                .AppendLine()
+                .AppendLine()
+                .AppendLine("### Excessive permissions");
+
+            _ = result.UsesMinimalPermissions
+                ? sb.AppendLine()
+                    .AppendLine("The token has the minimal permissions required.")
+                : sb.AppendLine()
+                    .AppendLine("The following permissions included in the token are unnecessary:")
+                    .AppendLine()
+                    .AppendJoin(Environment.NewLine, result.ExcessivePermissions.Select(p => $"- `{p}`"))
+                    .AppendLine();
+
+            _ = sb.AppendLine();
+        }
+
+        if (UnmatchedRequests.Any())
+        {
+            _ = sb.AppendLine("## Unmatched Requests")
+                .AppendLine()
+                .AppendLine("The following requests could not be matched:")
+                .AppendLine()
+                .AppendJoin(Environment.NewLine, UnmatchedRequests.Select(r => $"- {r}"))
+                .AppendLine()
+                .AppendLine();
+        }
+
+        if (Errors.Any())
+        {
+            _ = sb.AppendLine("## Errors")
+                .AppendLine()
+                .AppendLine("| Request | Error |")
+                .AppendLine("| ----------| ------------- |")
+                .AppendJoin(Environment.NewLine, Errors.Select(error => $"| {error.Request} | {error.Error} |"))
+                .AppendLine()
+                .AppendLine();
+        }
+
+        if (ExcludedPermissions?.Any() == true)
+        {
+            _ = sb.AppendLine("## Excluded Permissions")
+                .AppendLine()
+                .AppendLine("The following permissions were excluded from the analysis:")
+                .AppendLine()
+                .AppendJoin(Environment.NewLine, ExcludedPermissions.Select(p => $"- `{p}`"))
+                .AppendLine();
+        }
+
+        return sb.ToString();
+    }
+
+    public string? ToPlainText()
+    {
+        if (!Results.Any() && !UnmatchedRequests.Any() && !Errors.Any() && ExcludedPermissions?.Any() != true)
+        {
+            return "No permissions information to report.";
+        }
+
+        var sb = new StringBuilder();
+        _ = sb.AppendLine("Minimal Permissions Report")
+              .AppendLine("==========================");
```

Comment:
Nice touch! We should do this for other text reports. I'll add an empty line after each heading decorator to make the content more readable.